datadog/resource_datadog_screenboard.go

```diff
@@ -741,6 +853,96 @@ func buildTileDefRequestsConditionalFormats(source interface{}) []datadog.Condit
 	return r
 }
 
+func buildTileDefRequestsGroupBys(source interface{}) []datadog.TileDefApmOrLogQueryGroupBy {
+	groupBys, ok := source.([]interface{})
+	if !ok || len(groupBys) == 0 {
+		return nil
+	}
+
+	r := []datadog.TileDefApmOrLogQueryGroupBy{}
+	for _, groupBy := range groupBys {
+		groupByMap := groupBy.(map[string]interface{})
+		d := datadog.TileDefApmOrLogQueryGroupBy{}
+
+		batchSetFromDict(batch{
+			dict: groupByMap,
+			matches: []match{
+				{"facet", &d.Facet},
+				{"limit", &d.Limit},
+			}})
+		if groupBySort, ok := groupByMap["sort"].(map[string]interface{}); ok {
+			s := datadog.TileDefApmOrLogQueryGroupBySort{}
+			if aggr, ok := groupBySort["aggregation"].(string); ok && len(aggr) > 0 {
+				s.Aggregation = datadog.String(aggr)
+			}
+			if order, ok := groupBySort["order"].(string); ok && len(order) > 0 {
+				s.Order = datadog.String(order)
+			}
+			if facet, ok := groupBySort["facet"].(string); ok && len(facet) > 0 {
+				s.Facet = datadog.String(facet)
+			}
+			d.Sort = &s
+		}
+		r = append(r, d)
+	}
+	return r
+}
+
+func buildTileDefRequestsApmOrLogQuery(source interface{}) *datadog.TileDefApmOrLogQuery {
+	datadogQuery := source.(map[string]interface{})
+
+	// Index
+	d := datadog.TileDefApmOrLogQuery{
+		Index: datadog.String(datadogQuery["index"].(string)),
+	}
+
+	// Compute
+	terraformCompute := datadogQuery["compute"].(map[string]interface{})
+	datadogCompute := datadog.TileDefApmOrLogQueryCompute{}
+	if aggr, ok := terraformCompute["aggregation"].(string); ok && len(aggr) != 0 {
+		datadogCompute.Aggregation = datadog.String(aggr)
```

Comment:
Should we use the SetX methods from the datadog accessors for these?

Reply:
Not sure the history, but I notice we aren't really using any of the GetX or SetX methods in these resources.